src/pyvesync/vesyncoutlet.py

```diff
@@ -19,52 +19,76 @@ def __init__(self, details, manager):
     @abstractmethod
     def update(self):
         """Gets Device Energy and Status"""
-        raise NotImplementedError
+
+    @abstractmethod
+    def update_energy(self):
+        """Builds weekly, monthly and yearly dictionaries"""
 
     @abstractmethod
     def turn_on(self):
         """Return True if device has beeeen turned on"""
-        raise NotImplementedError
 
     @abstractmethod
     def turn_off(self):
         """Return True if device has beeeen turned off"""
-        raise NotImplementedError
 
     @abstractmethod
+    def get_details(self):
+        """Build details dictionary"""
+    
+    @abstractmethod
+    def get_weekly_energy(self):
+        """Build weekly energy history dictionary"""
+    
+    @abstractmethod
+    def get_monthly_energy(self):
+        """Build Monthly Energy History Dictionary"""
+
+    @abstractmethod
+    def get_yearly_energy(self):
+        """Build Yearly Energy Dictionary"""
+
     def active_time(self):
         """Return active time of a device in minutes"""
+        if 'active_time' not in self.details:
+            self.get_details()
```

Comment:
I learned a lesson on doing what you are trying to do here awhile ago. The HA team doesn't like it when anything that is expected to simply read and return a value makes a call out to an API to get the data if it doesn't exist. The library used to do exactly what you're doing here and since the HA integration was a passthrough to get the energy_today, power, and these types of values a hasty PR was made against the HA vesync integration that bypassed the library function that was causing updates

Reply:
Your right, I remember reading the convo in your original PR. I'll fix it